codec/src/main/java/io/netty/handler/codec/compression/Snappy.java

```diff
@@ -495,7 +495,7 @@ private static int decodeCopyWith2ByteOffset(byte tag, ByteBuf in, ByteBuf out,
 
         int initialIndex = out.writerIndex();
         int length = 1 + (tag >> 2 & 0x03f);
-        int offset = in.readShortLE();
+        int offset = in.readUnsignedShortLE();
```

Comment:
The source text is ambiguous on whether these are signed or not, but given it wouldn't make sense to have an offset that looked ahead of the current block I think it is reasonable to assume these are unsigned 16-bit integers.